src/discord/commands/book.ts

```diff
@@ -0,0 +1,227 @@
+        sectionId: sectionInformation._id,
+        startsAt: { $gte: startDate.toJSDate() },
+        endsAt: { $lte: endDate.toJSDate() },
+    });
+
+    const timeBlocks = [];
+    let currHourStart = startDate;
+    let currHourEnd = currHourStart.plus({ hours: 1 });
+
+    // Iterate through the hours in the given time range
+    while (currHourStart < endDate) {
+        // Finds schedule start and end for the day that the current hour falls on
+        const currWeekDay = currHourStart.weekday;
+        let scheduleDayStart = 0;
+        let scheduleDayEnd = 0;
+        for (const day of roomInformation.schedule) {
+            if (day.dayOfWeek + 1 === currWeekDay) {
+                scheduleDayStart = day.start;
+                scheduleDayEnd = day.end;
+            }
+        }
+
+        // Checks if the hour falls under an open time for the room and if it does adds hour to the list of available times
+        if (!roomInformation.closed && currHourStart.weekday === currHourEnd.weekday && currHourStart.hour >= scheduleDayStart && currHourEnd.hour <= scheduleDayEnd) {
+            // Check for time block in database
+            const bookedTimeBlockFound = bookedTimeblocks.find((bookedTimeBlock) => bookedTimeBlock.startsAt.getTime() === currHourStart.toMillis());
+
+            if (bookedTimeBlockFound) {
+                const newTimeBlock = {
+                    startsAt: currHourStart.toJSDate(),
+                    endsAt: currHourEnd.toJSDate(),
+                    availableCapacity: sectionInformation.capacity - bookedTimeBlockFound.users.length,
+                };
+                timeBlocks.push(newTimeBlock);
+            } else {
+                const newTimeBlock = {
+                    startsAt: currHourStart.toJSDate(),
+                    endsAt: currHourEnd.toJSDate(),
+                    availableCapacity: sectionInformation.capacity,
+                };
+                timeBlocks.push(newTimeBlock);
+            }
+        }
+
+        // Increments hour start and end for next iteration
+        currHourStart = currHourStart.plus({ hours: 1 });
+        currHourEnd = currHourEnd.plus({ hours: 1 });
+    }
+    return timeBlocks;
+}
+
+function timeConversion(timeObject: Date) {
+    //Converts 24 hour time to 12 hour time with a.m. and p.m. and changes 0:00 to 12:00
+    const currentHour = DateTime.fromJSDate(timeObject).hour;
+    const convertedTime = currentHour % 12 || 12;
+    const suffix = currentHour < 12 ? 'am' : 'pm';
+
+    return `${convertedTime}:00 ${suffix}`;
+}
+
+function parseTimeblocks(timeBlocks: TimeblockInformation[]) {
+    //Sets up select menu options for timeblocks
+    const timeBlockOptions = [];
+    for (const timeBlock of timeBlocks) {
+        timeBlockOptions.push({
+            label: `${timeConversion(timeBlock.startsAt)} - ${timeConversion(timeBlock.endsAt)}`,
+            value: `${timeConversion(timeBlock.startsAt)} - ${timeConversion(timeBlock.endsAt)}`,
```

Comment:
Does timeConversion handle time zones correctly? You could use Luxon to parse and format it:
```ts
DateTime.fromJSDate(timeBlock.startsAt, { zone: 'America/Toronto' }).toFormat('h:mm a')
```